src/api.v2/events/handleWorkRequest.js

```diff
@@ -1,62 +1,42 @@
-const getLogger = require('../../utils/getLogger');
+const generateETag = require('../helpers/worker/generateEtag');
 const getWorkResults = require('../helpers/worker/getWorkResults');
 
-const logger = getLogger();
 const validateAndSubmitWork = require('./validateAndSubmitWork');
 
-const handleWorkRequest = async (socket, data) => {
-  const { Authorization, experimentId, ETag } = data;
 
+//
+const getSignedUrlIfAvailable = async (experimentId, ETag) => {
```

Comment:
I saw there was a good explanation in the UI in which cases we wouldn't have anything to download, could you add it here too?

Reply:
not sure which one you are referring to, the cases are already commented in the code